### PR TITLE
Feature/add comment for directory

### PR DESCRIPTION
**Pull request by aliiohs**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: registry/directory/directory.go [hunk 1]
```diff
@@ -334,6 +338,7 @@ func (l *consumerConfigurationListener) addNotifyListener(listener registry.Noti
 	l.listeners = append(l.listeners, listener)
 }
 
+// Process handle events and update Invokers
```
Comment (AlexStocks) [inline registry/directory/directory.go:341]: it should be : Process handles events from xxx and updates Invokers

File: registry/directory/directory.go [hunk 2]
```diff
@@ -42,12 +42,12 @@ import (
 	"github.com/apache/dubbo-go/remoting"
 )
 
-// Options ...
+// Options Option is configuration related  struct of serviceTTL
 type Options struct {
```
Comment (zouyx) [inline registry/directory/directory.go:46]: what abount change to `Option`?

File: registry/directory/directory.go [hunk 3]
```diff
@@ -42,12 +42,12 @@ import (
...
 	serviceTTL time.Duration
 }
 
-// Option ...
+// Option function of handling Options
 type Option func(*Options)
```
Comment (zouyx) [inline registry/directory/directory.go:51]: what abount change to `OptionHandler` ?

File: registry/directory/directory.go [hunk 4]
```diff
@@ -90,18 +90,19 @@ func NewRegistryDirectory(url *common.URL, registry registry.Registry, opts ...O
 	return dir, nil
 }
 
-//subscribe from registry
+// Subscribe subscribe from registry
```
Comment (zouyx) [inline registry/directory/directory.go:93]:
```suggestion
// Subscribe from registry
```

File: registry/directory/directory.go [hunk 5]
```diff
@@ -256,7 +257,8 @@ func (dir *registryDirectory) List(invocation protocol.Invocation) []protocol.In
 	return routerChain.Route(invokers, dir.cacheOriginUrl, invocation)
 }
 
-func (dir *registryDirectory) IsAvailable() bool {
+// IsAvailable  whether the directory is available
+func (dir *registryDirectory) IsAvailableAvailable() bool {
```
Comment (zouyx) [inline registry/directory/directory.go:261]: why `IsAvailableAvailable`.........

File: registry/directory/directory.go (hunk 2 above)
Comment (hxmhlt) [reply registry/directory/directory.go:46]: The comment must be a complete sentence.

Comment (hxmhlt) [reply registry/directory/directory.go:46]: No `Options Option is configuration related  struct of serviceTTL` but `Options  is configuration related  struct of serviceTTL`

File: registry/directory/directory.go (hunk 3 above)
Comment (hxmhlt) [reply registry/directory/directory.go:51]: Option function of handling Options => Option will define a function of handling Options.

File: registry/directory/directory.go [hunk 6]
```diff
@@ -66,7 +66,7 @@ type registryDirectory struct {
 	forbidden  atomic.Bool
 }
 
-// NewRegistryDirectory ...
+// NewRegistryDirectory  Create a new RegistryDirectory
 func NewRegistryDirectory(url *common.URL, registry registry.Registry, opts ...Option) (*registryDirectory, error) {
```
Comment (hxmhlt) [inline registry/directory/directory.go:70]: NewRegistryDirectory  Create a new RegistryDirectory => NewRegistryDirectory  will create a new RegistryDirectory

**Comment by flycash** [comment]
Please resolve conflict

**Pull request merged**